Source/ACE/Entity/WorldObject.cs

```diff
@@ -80,59 +82,184 @@ public List<AceObjectPropertiesSpell> PropertiesSpellId
             get { return AceObject.SpellIdProperties; }
         }
 
-        public ItemType ItemType
+        #region ObjDesc
+        private readonly List<ModelPalette> modelPalettes = new List<ModelPalette>();
+
+        private readonly List<ModelTexture> modelTextures = new List<ModelTexture>();
+
+        private readonly List<Model> models = new List<Model>();
+
+        // subpalettes
+        public List<ModelPalette> GetPalettes
         {
-            get { return (ItemType?)AceObject.ItemType ?? 0; }
-            protected set { AceObject.ItemType = (uint)value; }
+            get { return modelPalettes.ToList(); }
         }
 
-        /// <summary>
-        /// wcid - stands for weenie class id
-        /// </summary>
-        public uint WeenieClassId
+        // tmChanges
+        public List<ModelTexture> GetTextures
         {
-            get { return AceObject.WeenieClassId; }
-            protected set { AceObject.WeenieClassId = value; }
+            get { return modelTextures.ToList(); }
         }
 
-        public uint? IconId
+        // apChanges
+        public List<Model> GetModels
         {
-            get { return AceObject.IconDID; }
-            set { AceObject.IconDID = value; }
+            get { return models.ToList(); }
         }
 
-        public string Name
+        public void AddPalette(uint paletteId, ushort offset, ushort length)
         {
-            get { return AceObject.Name; }
-            protected set { AceObject.Name = value; }
+            ModelPalette newpalette = new ModelPalette(paletteId, offset, length);
+            modelPalettes.Add(newpalette);
         }
 
-        public IActor CurrentParent { get; private set; }
+        public void AddTexture(byte index, uint oldtexture, uint newtexture)
+        {
+            ModelTexture nextTexture = new ModelTexture(index, oldtexture, newtexture);
+            modelTextures.Add(nextTexture);
+        }
 
-        public Position ForcedLocation { get; private set; }
+        public void AddModel(byte index, uint modelresourceid)
+        {
+            Model newmodel = new Model(index, modelresourceid);
+            models.Add(newmodel);
+        }
 
-        public Position RequestedLocation { get; private set; }
+        public void Clear()
```

Comment:
Either fix, or comment what it does (to be later fixed then), its a deceptive name.